parsl/monitoring/app_monitor.py

```diff
@@ -5,59 +5,87 @@
-    logger.info("starting monitoring for {} on {}".format(pid, os.getpid(), extra={'task_id': task_id, 'task_run_id': run_id}))
     pm = psutil.Process(pid)
     pm.cpu_percent()
 
     def to_mb(some_bytes):
         return some_bytes / 1024**2
 
     while True:
-        children = pm.children(recursive=True)
-        d = {"psutil_process_" + str(k): v for k, v in pm.as_dict().items() if k in simple}
-        d["psutil_cpu_count"] = psutil.cpu_count()
-        d["task_run_id"] = run_id
-        d["task_id"] = task_id
-        d['psutil_process_memory_virtual'] = to_mb(pm.memory_info().vms)
-        d['psutil_process_memory_resident'] = to_mb(pm.memory_info().rss)
-        d['psutil_process_time_user'] = pm.cpu_times().user
-        d['psutil_process_time_system'] = pm.cpu_times().system
-        d['psutil_process_children_count'] = len(children)
         try:
-            d['psutil_process_disk_write'] = to_mb(pm.io_counters().write_bytes)
-            d['psutil_process_disk_read'] = to_mb(pm.io_counters().read_bytes)
-        except psutil._exceptions.AccessDenied:
-            # not setting should result in a null value that should report as a blank and not "spoil" the kibana aggregations
-            pass
-        for child in children:
-            for k, v in child.as_dict(attrs=summable_values).items():
-                d['psutil_process_' + str(k)] += v
-            d['psutil_process_time_user'] += child.cpu_times().user
-            d['psutil_process_time_system'] += child.cpu_times().system
-            d['psutil_process_memory_virtual'] += to_mb(child.memory_info().vms)
-            d['psutil_process_memory_resident'] += to_mb(child.memory_info().rss)
+            d = {"psutil_process_" + str(k): v for k, v in pm.as_dict().items() if k in simple}
+            d["task_run_id"] = run_id
+            d["task_id"] = task_id
+            children = pm.children(recursive=True)
+            d["psutil_cpu_count"] = psutil.cpu_count()
+            d['psutil_process_memory_virtual'] = to_mb(pm.memory_info().vms)
+            d['psutil_process_memory_resident'] = to_mb(pm.memory_info().rss)
+            d['psutil_process_time_user'] = pm.cpu_times().user
+            d['psutil_process_time_system'] = pm.cpu_times().system
+            d['psutil_process_children_count'] = len(children)
             try:
-                d['psutil_process_disk_write'] += to_mb(child.io_counters().write_bytes)
-                d['psutil_process_disk_read'] += to_mb(child.io_counters().read_bytes)
+                d['psutil_process_disk_write'] = to_mb(pm.io_counters().write_bytes)
+                d['psutil_process_disk_read'] = to_mb(pm.io_counters().read_bytes)
             except psutil._exceptions.AccessDenied:
-                pass
-        logger.info("test", extra=d)
-        time.sleep(5)
+                # occassionally pid temp files that hold this information are unvailable to be read so set to zero
```

Comment:
is this a bug? in this code? in some other code?

Reply:
Good question. I do not know the answer for exactly why this happens but from my experience, it seems that psutil is not always successful at gathering certain information about every process. This is one of a few unreliable points in the collection process. My approach was that it was relatively harmless to take this simple approach of ignoring failed attempts to gather the information and take what we can get when it is available with the assumption that over the course of a workflow the general idea of what is happening will still be clear. It is likely useful to explore through testing how accurate this resource collecting process is. As far as I can tell it seems rather representative of what I would expect but I do not have any hard evidence for its performance or accuracy.